subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -349,6 +361,68 @@ static struct bass_recv_state_internal *bass_lookup_addr(const bt_addr_le_t *add
 	return NULL;
 }
 
+#if defined(CONFIG_BT_BAP_SCAN_DELEGATOR_OVERWRITE_OLDEST)
+static struct bass_recv_state_internal *free_oldest_inactive_state(void)
+{
+	/* Free the oldest updated receive state that is not currently being synced */
+
+	LOG_DBG("Looking for oldest inactive state to free");
+
+	struct bass_recv_state_internal *oldest_inactive_state = NULL;
+
+	for (size_t i = 0U; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		struct bass_recv_state_internal *current_state =
+					&scan_delegator.recv_states[i];
+
+		struct bt_bap_scan_delegator_recv_state *state = &current_state->state;
+
+		if (IS_ENABLED(CONFIG_BT_BAP_BROADCAST_SINK)) {
+			/* First check that if the receive state is associated with a
+			 * broadcast sink, that it is not currently synced or syncing
+			 * to a big
+			 */
+			struct bt_bap_broadcast_sink *broadcast_sink =
+					bt_bap_broadcast_sink_get_by_bass_src_id(state->src_id);
+
+			if (broadcast_sink != NULL && broadcast_sink->big != NULL) {
+				/* BIG is being synced */
+				LOG_DBG("Receive state at index %zu has an associated big. "
+						"Unable to free", i);
+				continue;
+			}
+		}
+
+		/* Now check if the receive state is currently pa_synced */
+		if (state->pa_sync_state == BT_BAP_PA_STATE_SYNCED) {
+			LOG_DBG("Receive state at index %zu is PA synced. "
+					"Unable to free", i);
+			continue;
+		}
+
+		LOG_DBG("Receive state at index %zu is currently inactive", i);
+
+		if ((oldest_inactive_state == NULL) ||
+			(current_state->aging_counter < oldest_inactive_state->aging_counter)) {
+			oldest_inactive_state = current_state;
+		}
+	}
+
+	if (oldest_inactive_state != NULL) {
+		LOG_DBG("Receive state at index %zu is the oldest inactive receive state. "
+				"This will be used", oldest_inactive_state->index);
+
+		oldest_inactive_state->active = false;
+		oldest_inactive_state->pa_sync = NULL;
+		memset(&oldest_inactive_state->state, 0, sizeof(oldest_inactive_state->state));
+		memset(&oldest_inactive_state->broadcast_code, 0, sizeof(oldest_inactive_state->broadcast_code));
+		memset(&oldest_inactive_state->requested_bis_sync, 0, sizeof(oldest_inactive_state->requested_bis_sync));
```

Comment:
We may want to implement a static function to reset receive states so that we don't have these lines multiple places